skyportal/tests/flaky/test_frontend_flaky.py

```diff
@@ -243,3 +243,63 @@ def test_super_user_post_allocation(
 
     # check for dropdown instrument
     driver.wait_for_xpath(f'//span[text()[contains(.,"{instrument_name2}/{name}")]]')
+
+
+@pytest.mark.flaky(reruns=3)
+def test_super_user_post_shift(
+    public_group, super_admin_token, super_admin_user, driver
+):
+
+    driver.get(f"/become_user/{super_admin_user.id}")
+
+    # go to the shift page
+
+    name = str(uuid.uuid4())
+    status, data = api(
+        'POST',
+        'shift',
+        data={
+            'name': name,
+            'group_id': public_group.id,
+            'start_date': '3021-02-27T00:00:00',
+            'end_date': '3021-07-20T00:00:00',
+        },
+        token=super_admin_token,
+    )
+    assert status == 200
+    assert data['status'] == 'success'
+
+    driver.get("/shifts")
+
+    # check for API shift
+    driver.wait_for_xpath(f'//*[text()[contains(.,"{name}")]]', timeout=20)
+
+    form_name = str(uuid.uuid4())
+    driver.wait_for_xpath('//*[@id="root_name"]').send_keys(form_name)
+    driver.wait_for_xpath('//*[@id="root_start_date"]').send_keys('01/01/2022')
+    driver.wait_for_xpath('//*[@id="root_start_date"]').send_keys(Keys.TAB)
+    driver.wait_for_xpath('//*[@id="root_start_date"]').send_keys('01:01')
+    driver.wait_for_xpath('//*[@id="root_start_date"]').send_keys('P')
+
+    driver.wait_for_xpath('//*[@id="root_end_date"]').send_keys('03/01/2022')
+    driver.wait_for_xpath('//*[@id="root_end_date"]').send_keys(Keys.TAB)
+    driver.wait_for_xpath('//*[@id="root_end_date"]').send_keys('01:01')
+    driver.wait_for_xpath('//*[@id="root_end_date"]').send_keys('P')
+
+    driver.wait_for_xpath('//*[@id="root_end_date"]').send_keys('02/01/2022 01:01:10')
+
+    driver.click_xpath('//*[@id="root_group_id"]')
+    driver.click_xpath('//li[contains(text(), "Sitewide Group")]')
+
+    submit_button_xpath = '//button[@type="submit"]'
+    driver.wait_for_xpath(submit_button_xpath)
+    driver.click_xpath(submit_button_xpath)
+
+    # check for dropdown shift
+    driver.wait_for_xpath(f'//span[text()[contains(.,"{form_name}")]]')
+
+    # check for delete shift button
+    delete_button_xpath = '//button[@id="delete_button"]'
```

Comment:
This will find a delete button, but not necessarily the one for the right shift (if there are multiple shifts). Maybe this also makes this test flaky. Probably need to make the xpath more specific to the one shift with the right name.

Reply:
Done. Now looks for the list element that contains the same group name and shift name as the shift we just added